infrastructure/gsuite-exporter/gsuite_exporter/exporters/stackdriver_exporter.py

```diff
@@ -0,0 +1,143 @@
+
+    def send(self, records, log_name, dry=False):
+        """Writes a list of Admin SDK records to Stackdriver Logging API.
+
+        Args:
+            records (list): A list of log records.
+            log_name (str): The log name to write (e.g: 'logins').
+            dry (bool): Toggle dry-run mode (default: False).
+
+        Returns:
+            `googleapiclient.http.HttpRequest`: The API response object.
+        """
+        res = None
+        destination = self.get_destination(log_name)
+        if records:
+            entries = self.convert(records)
+            body = {
+                'entries': entries,
+                'logName': '{}'.format(destination),
+                'dryRun': dry
+            }
+            logger.debug("Writing {} entries to Stackdriver Logging API @ '{}'".format(
+                len(entries),
+                destination))
+            res = self.api.entries().write(body=body).execute()
+        return res
+
+    def convert(self, records):
+        """Convert a bunch of Admin API records to Stackdriver Logging API
+        entries.
+
+        Args:
+            records (list): A list of Admin API records.
+
+        Returns:
+            list: A list of Stackdriver Logging API entries.
+        """
+        return map(lambda i: self.__convert(i), records)
+
+    def get_destination(self, log_name):
+        return "{}/logs/{}".format(self.project_id, log_name)
+
+    def get_last_timestamp(self, log_name):
+        """Last log timestamp from Stackdriver Logging API given our project id
+        and log name.
+        """
+        destination = self.get_destination(log_name)
+        query = {
+          'orderBy': 'timestamp desc',
+          'pageSize': 1,
+          'resourceNames': [self.project_id],
+          'filter': 'logName={}'.format(destination)
+        }
+        log = self.api.entries().list(body=query).execute()
+        timestamp = log['entries'][0]['timestamp'] if 'entries' in log else None
+        return timestamp
+
+    def __convert(self, record):
+        """Converts an Admin SDK log entry to a Stackdriver Log entry.
+
+        Args:
+            record (dict): The Admin SDK record as JSON.
+
+        Returns:
+            dict: The Stackdriver Logging entry as JSON.
+        """
+        return {
+            'timestamp': {'seconds': int(time.time())},
+            'insertId': record['etag'],
+            'jsonPayload': {
+                'requestMetadata': {'callerIp': record.get('ipAddress')},
+                'authenticationInfo': {
+                    'callerType': record['actor'].get('callerType'),
+                    'principalEmail': record['actor'].get('email')
+                },
+                'methodName': record['events'][0]['name'],
+                'parameters': record['events'][0].get('parameters'),
+                'report_timestamp': self.__convert_timestamp(record)
+            },
+            'resource': { 'type': 'global' }
```

Comment:
Google Python standards discourage whitespace in braces

Reply:
Fixed.